### DISCUSSION TITLE
ESP32

### DISCUSSION THREAD

**Discussion by Hamish1066** [TheRealWaldo/thermal#184] [Ideas]
Firstly, thank you so much for making this. It’s awesome, and I’m currently running it. I want to make more of these, but I want to do it on an ESP32 so that I can connect them via ethernet. I’m new to esphome and coding - can I just use the same code with and ESP32? 

Thanks again for making this. I hope it becomes an official integration. It’s really great, and I think it’s a way cooler and better solution to motion sensing compared to mmWave. I want to make one of these for every room in the house, but want to learn more before diving too deeply into it.

**Comment by TheRealWaldo** [comment]
@Hamish1066 generally, yes, with minor modifications.  A note that this repository Is solely for the Home Assistant Integration.  You will have better luck visiting the project pages for the various hardware as you may find someone who has already built a version for your board.

See https://github.com/TheRealWaldo/AMG8833-ESPHOME/discussions/10 for an example.

**Comment by Hamish1066** [comment]
Thanks! I'll poke around some more. Thank you for the links. 

I had another question, and can start a new issue if you prefer. I've been running your code on my D1 mini 8266 for a few months now. It was working perfectly, but recently started going from person detected / clear every few seconds. I tried playing around with line 42 in the person detected.h file - I went from 1 - 6 for the sensitivity - but there were no changes. The only thing that corrected it was restarted the device a few times. 

Is line 42 the correct way to adjust the sensitivity, or was I incorrect in assuming this? Have you encountered this issue and found any fixes?

**Reply by TheRealWaldo** [reply]
So that line represents the 'difference' between max and min temperatures.  A higher value would mean you expect the ambient temperature to be much lower than what you are detecting.

The 'sensitivity,' if you will, is based on the quality of the sensor itself.  A challenge with the AMG8833 is its lower resolution.  The further away from the sensor, the more likely a warmer surface will be 'merged into' a cool surface as a single pixel.

So if you're having issues, it could be due to positioning.

That being said, a reboot correcting it makes me think that there may be a memory leak or problem with the host board itself.  It's not likely to be a bug in the code.
